File: Niva/NivaInNiva/.niva_go/testss/testss.go
package main

import (
    "niva/core"
)

type Person struct {
  Name string
  Age int
}

func Person__hello(this *Person) {
  name := this.Name
  _ = name
  age := this.Age
  _ = age
  core.Any__echo(((("hello my name is 123 ") + (name))))
}


func main() {
  p := &Person{Name: "Alice", Age: 24}
  _ = p
  Person__hello(p)
  list := []int{ 1, 2, 3 }
  _ = list
  hashSet := map[int]struct{}{1: struct{}{}, 2: struct{}{}, 3: struct{}{}}
  _ = hashSet
  hashMap := map[int]int{1: 2, 3: 4}
  _ = hashMap
  core.Any__echo(list)
  core.Any__echo(hashSet)
  core.Any__echo(hashMap)
  core.List__forEach((core.List__filter((core.List__map(list, func(it int) (result int) {
    return ((it) * (2))
  })), func(it int) (result bool) {
    return ((((it) % (2))) == (0))
  })), func(it int) {
    core.Any__echo(it)
  })
  _ = core.Bool__ifTrue__ifFalse(((1) > (2)), func() (result int) {
    return 3
  }, func() (result int) {
    return 4
  })
  _ = (((((1) > (2)))) && ((((2) > (1)))))
}

File: Niva/NivaInNiva/.niva_js/testss.js
import * as core from "./core.js"


export class Person {
    constructor(name,age) {
        this.name = name;
        this.age = age;
    }
}


export function Person__hello(_this) {
    let name = _this.name
    let age = _this.age
    core.Any__echo(((("hello my name is 3212 ") + (name))))
}


const p = new Person("Alice", 24);
Person__hello(p);
const list = [ 1, 2, 3 ];
const hashSet = new Set([1, 2, 3]);
const hashMap = new Map([[1, 2], [3, 4]]);
core.Any__echo(list);
core.Any__echo(hashSet);
core.Any__echo(hashMap);
core.List__forEach((core.List__filter((core.List__map(list, (it) => {return ((it) * (2))})), (it) => {return ((((it) % (2))) === (0))})), (it) => {return core.Any__echo(it)});